### DISCUSSION TITLE
Bank&program change via USB

### DISCUSSION THREAD

**Discussion by anycolorsmith** [probonopd/MiniDexed#255] [Q&A]
I can change bank, voice and volume from my midi keyboard for all TG together, what other parameters can be changed in this way?

**Comment by probonopd** [comment]
All voice parameters should be possible to change in this way, if your keyboard can send DX7 style sysex messages.

Once https://github.com/probonopd/MiniDexed/issues/63 gets implemented, you should be able to use also keyboards that cannot send DX7 style sysex messages.

**Reply by anycolorsmith** [reply]
For all generators at once i think?

**Reply by probonopd** [reply]
MIDI CC messages: For the Tone Generators set to the channel on which the MIDI controller is sending messages, and for the Tone Generators set to "Omni".

MIDI sysex messages: Potentially only for one TG, we are still figuring out how this should work. [Details](https://github.com/probonopd/MiniDexed/issues/82#issuecomment-1128232574)

**Reply by anycolorsmith** [reply]
Thanx!!
